File: uncategorized/1503_last_moment_before_all_ants_fall_out_of_a_plank.js
/**
 * @param {number} n
 * @param {number[]} left
 * @param {number[]} right
 * @return {number}
 */
var getLastMoment = function(n, left, right) {

    let leftMax = (Math.max(... left) || 0)
    let rightMax = (Math.max(... right.map((num) => n-num)) || 0)
    return Math.max(leftMax, rightMax)

};

File: uncategorized/1503_last_moment_before_all_ants_fall_out_of_a_plank.go
func getLastMoment(n int, left []int, right []int) int {

 leftMax := 0
 rightMax := 0

 for _, num := range left {
    if leftMax < num {
        leftMax = num
    }
 }

 for _, num := range right {
     if rightMax < n-num {
         rightMax = n-num
     }
 }

ans := 0
 if ans < leftMax {
     ans = leftMax
 }
 if ans < rightMax {
     ans = rightMax
 }

 return ans
}